include/cactus_rt/cyclic_thread.h

```diff
@@ -0,0 +1,112 @@
+#ifndef CACTUS_RT_CYCLIC_THREAD_H_
+#define CACTUS_RT_CYCLIC_THREAD_H_
+
+#include <spdlog/spdlog.h>
+
+#include <atomic>
+#include <vector>
+
+#include "cactus_rt/latency_tracker.h"
+#include "cactus_rt/thread.h"
+#include "cactus_rt/utils.h"
+
+namespace cactus_rt {
+
+// TODO: Type assertion / concept to restrict SchedulerT to viable
+template <typename SchedulerT>
+class CyclicThread : public Thread<SchedulerT> {
+  int64_t                     period_ns_;
+  struct timespec             next_wakeup_time_;
+  typename SchedulerT::Config scheduler_config_;
+  std::atomic_bool            should_stop_ = false;
+
+  // Debug information
+  LatencyTracker wakeup_latency_tracker_;
+  LatencyTracker loop_latency_tracker_;
+  LatencyTracker busy_wait_latency_tracker_;
+
+ public:
+  /**
+   * Creates a thread that loops at a set period
+   *
+   */
+  CyclicThread(const std::string&                 name,
+               int64_t                            period_ns = 1'000'000,
+               const typename SchedulerT::Config& config = {},
+               std::vector<size_t>                cpu_affinity = {},
+               size_t                             stack_size = kDefaultStackSize)
+      : Thread<SchedulerT>(name, config, cpu_affinity, stack_size),
+        period_ns_(period_ns),
+        scheduler_config_(config) {
+  }
+
+  virtual ~CyclicThread() = default;
+
+  void RequestStop() noexcept {
+    should_stop_.store(true);
+  }
+
+ protected:
+  void Run() noexcept final {
+    clock_gettime(CLOCK_MONOTONIC, &next_wakeup_time_);
+    int64_t loop_start, loop_end, should_have_woken_up_at;
+
+    double wakeup_latency, loop_latency, busy_wait_latency;
```

Comment:
```suggestion
    int64_t wakeup_latency, loop_latency, busy_wait_latency;
```

Now that this is all in nanoseconds, there's no reason to use double.